sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -376,19 +284,15 @@ export class MessageReceiver extends LinkEntity {
    * @return {Promise<void>} Promise<void>.
    */
   async close(): Promise<void> {
-    this.wasCloseInitiated = true;
     log.receiver(
       "[%s] Closing the [%s]Receiver for entity '%s'.",
       this._context.connectionId,
       this.receiverType,
       this._entityPath
     );
     this._clearAllMessageLockRenewTimers();
-    if (this._receiver) {
-      const receiverLink = this._receiver;
-      this._deleteFromCache();
-      await this._closeLink(receiverLink);
-    }
+    this._deleteFromCache();
+    await this.closeLink("permanently");
```

Comment:
We are calling `_deleteFromCache()` only when `closeLink()` is called with `permanently`, which makes sense. Closing the class permanently would mean we remove it from cache.

So, for a future PR, consider pulling the removal from cache inside `closeLink()` based on whether `permanently` was passed or not

Reply:
Mentioned in #10656